src/backend/distributed/planner/distributed_planner.c

```diff
@@ -2321,6 +2321,77 @@ HasUnresolvedExternParamsWalker(Node *expression, ParamListInfo boundParams)
 }
 
 
+/*
+ * IsExternParamUsedInQuery returns true if the passed in paramId
+ * is used in the query, false otherwise.
+ */
+bool
+GetParamsUsedInQuery(Node *expression, Bitmapset **paramBitmap)
```

Comment:
Would be nice to move these to something like utils/params.c

Reply:
Done.